ReSharper.FSharp/src/FSharp.Psi.Features/src/ExtendSelection/FSharpExtendSelectionProvider.fs

```diff
@@ -117,13 +128,28 @@ type FSharpExtendSelectionProvider(settingsStore: ISettingsStore) =
                 if bindingIndex >= bindings.Count then null else
 
                 FSharpExtendSelectionProvider.CreateLetBindingSelection(fsFile, letExpr, bindings.[bindingIndex])
+                
+            | :? ITypeDeclarationGroup as declGroup ->
+                let declarations = declGroup.TypeDeclarations
+                if declarations.IsEmpty then null else
+                
+                let tokenType = getTokenType token
+                if tokenType == FSharpTokenType.TYPE then
+                    FSharpExtendSelectionProvider.CreateTypeDeclarationSelection(fsFile, declGroup, declarations.[0]) else
+
+                if tokenType != FSharpTokenType.AND then null else
+
+                let declIndex = declGroup.Separators.IndexOf(token) + 1
+                if declIndex <= 0 || declIndex >= declarations.Count then null else
+
+                FSharpExtendSelectionProvider.CreateTypeDeclarationSelection(fsFile, declGroup, declarations.[declIndex])
```

Comment:
Could this code be unified with with let binding selection? Seems like it's really duplicating the logic.